autotests/testolmaccount.cpp

```diff
@@ -0,0 +1,583 @@
+    });
+    alice->run(res);
+
+    auto olm1 = alice1->olmAccount();
+    olm1->generateOneTimeKeys(10);
+    auto res1 = olm1->createUploadKeyRequest(olm1->oneTimeKeys());
+    QSignalSpy spy1(res1, &BaseJob::result);
+    connect(res1, &BaseJob::result, this, [res1] {
+        QCOMPARE(res1->oneTimeKeyCounts().size(), 1);
+        QCOMPARE(res1->oneTimeKeyCounts()["signed_curve25519"], 10);
+    });
+    alice1->run(res1);
+
+    auto olm2 = alice2->olmAccount();
+    olm2->generateOneTimeKeys(10);
+    auto res2 = olm2->createUploadKeyRequest(olm2->oneTimeKeys());
+    QSignalSpy spy2(res2, &BaseJob::result);
+    connect(res2, &BaseJob::result, this, [res2] {
+        QCOMPARE(res2->oneTimeKeyCounts().size(), 1);
+        QCOMPARE(res2->oneTimeKeyCounts()["signed_curve25519"], 10);
+    });
+    alice2->run(res2);
+
+
+    QVERIFY(spy.wait(10000));
+    QVERIFY(spy1.wait(10000));
+    QVERIFY(spy2.wait(1000)); // TODO this is failing even with 10000
+
+    // Bob will claim all keys from alice
+    CREATE_CONNECTION(bob, "bob", "secret", "BobPhone")
+
+    QStringList devices_;
+    devices_ << alice->deviceId()
+             << alice1->deviceId()
+             << alice2->deviceId();
+
+    QHash<QString, QHash<QString, QString>> oneTimeKeys;
+    for (const auto &d : devices_) {
+        oneTimeKeys[alice->userId()] = QHash<QString, QString>();
+        oneTimeKeys[alice->userId()][d] = SignedCurve25519Key;
+    }
+    auto job = bob->callApi<ClaimKeysJob>(oneTimeKeys);
+    connect(job, &BaseJob::result, this, [bob, job] {
+        const auto userId = bob->userId();
+        const auto deviceId = bob->deviceId();
+
+        // The device exists.
+        QCOMPARE(job->oneTimeKeys().size(), 1);
+        QCOMPARE(job->oneTimeKeys()[userId].size(), 3);
+    });
+}
+
+void TestOlmAccount::keyChange()
+{
+    CREATE_CONNECTION(alice, "alice", "secret", "AlicePhone")
+
+    auto job = alice->createRoom(Connection::PublishRoom, QString(), QString(), QString(), QStringList());
+    connect(job, &BaseJob::result, this, [alice, job, this] () {
+        // Alice syncs to get the first next_batch token.
+        alice->sync();
+        connect(alice.get(), &Connection::syncDone, this, [alice, this] {
+            const auto nextBatchToken = alice->nextBatchToken();
+
+            // generate keys and change existing one
+            auto aliceOlm = alice->olmAccount();
+            aliceOlm->generateOneTimeKeys(1);
+            auto aliceRes = aliceOlm->createUploadKeyRequest(aliceOlm->oneTimeKeys());
+            QSignalSpy spy(aliceRes, &BaseJob::result);
+
+            alice->run(aliceRes);
+            QVERIFY(spy.wait(10000));
+
+            // The key changes should contain her username
+            // because of the key uploading.
+
+            auto changeJob = alice->callApi<GetKeysChangesJob>(nextBatchToken, "");
+            connect(changeJob, &BaseJob::result, this, [&changeJob, &alice] {
```

Comment:
Those captured variables are pointers - there's no benefit in capturing them by reference. And were `QSignalSpy::wait()` not there, the captured references would go dangling.
```suggestion
            connect(changeJob, &BaseJob::result, this, [changeJob, alice] {
```